PlayerMovement.cs

```diff
@@ -184,87 +170,50 @@ public class PlayerMovement : MonoBehaviour {
         orientation.transform.localRotation = Quaternion.Euler(0, desiredX, 0);
     }
 
-    private void CounterMovement(float x, float y, Vector2 mag) {
-        if (!grounded || jumping) return;
+    void ApplyFriction() {
+        if (!grounded) return;
 
         //Slow down sliding
         if (crouching) {
-            rb.AddForce(moveSpeed * Time.deltaTime * -rb.velocity.normalized * slideCounterMovement);
+            rigidBody.AddForce(moveSpeed * Time.deltaTime * -rigidBody.velocity.normalized * slideFriction);
             return;
         }
 
-        //Counter movement
-        if (Math.Abs(mag.x) > threshold && Math.Abs(x) < 0.05f || (mag.x < -threshold && x > 0) || (mag.x > threshold && x < 0)) {
-            rb.AddForce(moveSpeed * orientation.transform.right * Time.deltaTime * -mag.x * counterMovement);
-        }
-        if (Math.Abs(mag.y) > threshold && Math.Abs(y) < 0.05f || (mag.y < -threshold && y > 0) || (mag.y > threshold && y < 0)) {
-            rb.AddForce(moveSpeed * orientation.transform.forward * Time.deltaTime * -mag.y * counterMovement);
+        Vector3 inverseVelocity = -transform.InverseTransformDirection(rigidBody.velocity);
```

Comment:
You can get the velocity relative to the transform this way, which removes the need for `FindVelRelativeToLook` which wasn't the most easy to understand